### DISCUSSION TITLE
Installation on Plesk server not possible - Composer plugins disabled

### DISCUSSION THREAD

**Discussion by ghost** [kimai/kimai#5015] [On-Premise: Installation & Updates]
Hello,

I tried to install Kimai on my machine running Plesk Obsidian by reaching out to the [official documentation](https://www.kimai.org/documentation/shared-hosting.html#plesk). Unfortunately, there's a problem installing the composer packages using Composer 2.7.7. When I run `/opt/plesk/php/8.1/bin/php ../composer.phar install --no-dev --optimize-autoloader -n` as described, I receive the following error message from Composer:

```
Composer plugins have been disabled for safety in this non interactive session.
Set COMPOSER_ALLOW_SUPERUSER=1 if you want to allow plugins to run as root/super user.
Do not run Composer as root/super user! See https://getcomposer.org/root for details.
```

I read a bit about the recommended line `COMPOSER_ALLOW_SUPERUSER=1` and figured out that it's already included in the `Dockerfile`, but it's still not working. I also don't know how to include this line into `composer.json` and I found nothing about it online.

When I'm trying to run the command with a non-root user, I get another error saying that I have no permission to modify Plesk files (also as described in the documentation).

Any ideas how this can be fixed?

Regards,
Niklas

**Comment by ghost** [comment]
@kevinpapst I think this is an issue that definitely needs attention as it probably affects all Plesk users.

**Reply by kevinpapst** [reply]
And I should care why? If Plesk is blocking the installation Kimai, talk to Plesk, not to Kimai.

Remove the `-n` flag and try again.

**Reply by ghost** [reply]
Please read my post again - it has nothing to do with Plesk itself. The documentation describes perfectly how to avoid problems with Plesk's file structure, even it's the "crowbar method" and worst option to install software as root. The problems occur when trying to install Composer packages. However, there is a fundamental problem with the installation within a Plesk environment.

The Plesk part in the installation section of the documentation is simply not up to date. I have solved the problem and could offer to update the documentation. But definitely not that way. 🤷🏻‍♂️

**Reply by kevinpapst** [reply]
You started by pinging me (forcing your way in to my inbox) and saying

> I think this is an issue that definitely needs attention

But it aint for me. Its important for you, because you want to use Kimai this way. 
If you can fix the documentation for the next user: great 👍 

> I have solved the problem and could offer to update the documentation. But definitely not that way.

This is a toxic statement to hurt me, after I give you my software for free 👎 

Here, I fixed the docs for you:
https://www.kimai.org/documentation/shared-hosting.html

<img width="770" alt="Bildschirmfoto 2024-08-08 um 11 30 14" src="https://github.com/user-attachments/assets/49cb4485-ba82-4acc-9693-aa10b4c16bbc">
